### PR TITLE
[cesium] added length check for series when updating high watermark

### PR DESCRIPTION
**Pull request by Lham42**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-964]()

## Description

Maximum 2-3 sentence description describing the changes.

Server would panic when disconnecting an NI device which is in use by a read task in the server.  Added a check to `UpdateHighWater` to check for series length before using it to set the watermark.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added sufficient regression tests to cover the changes.

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

### CODE REVIEW

File: cesium/writer_stream.go [hunk 1]
```diff
@@ -472,7 +472,15 @@ func (w *idxWriter) updateHighWater(s telem.Series) error {
 			s.DataType,
 		)
 	}
+	if s.Len() == 0 {
```
Comment (LeonLiur) [inline cesium/writer_stream.go:475]: Why should this error rather than doing nothing and just return?

File: x/go/telem/series.go [hunk 2]
```diff
@@ -55,6 +55,8 @@ func (s Series) Split() [][]byte {
 }
 
 func ValueAt[T types.Numeric](a Series, i int64) T {
```
Comment (LeonLiur) [inline x/go/telem/series.go:57]: I would rename the `a` variable to `s` for Series

**Review by LeonLiur — CHANGES_REQUESTED** [review]

File: cesium/writer_stream.go (hunk 1 above)
Comment (emilbon99) [reply cesium/writer_stream.go:475]: Agreed, we definitely shouldn't error here. If this series has length 0, then all other series in the frame must have length 0 per `validateWrite` so no data gets written. It honestly might be clearer to do this check just after the call to `validateWrite` on line 340 though. It feels more explicit that silently writing no data to all unary writers.

Comment (emilbon99) [reply cesium/writer_stream.go:475]: If we do choose to keep it here, we should add a test case that asserts its ok to write an empty series to a unary DB

**Review by emilbon99 — CHANGES_REQUESTED** [review]
Would be good to add a regression test that asserts we don't panic anymore

File: cesium/writer_stream.go (hunk 1 above)
Comment (Lham42) [reply cesium/writer_stream.go:475]: So if the series is empty do we still want to be able to write? Then would it make more sense just to add to the condition on line 351 to not update the watermark if the series is empty?
